### PR TITLE
TSK-145: add "domain" to Task

### PR DESCRIPTION
**Pull request by BVier**

### CODE REVIEW

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java [hunk 1]
```diff
@@ -400,6 +403,102 @@ public void testThrowsExceptionIfMandatoryPrimaryObjectReferenceIsNotSetOrIncomp
...
+
+        TaskService taskService = taskanaEngine.getTaskService();
+        WorkbasketService workbasketService = taskanaEngine.getWorkbasketService();
+
+        Workbasket wb1 = workbasketService.getWorkbasketByKey("USER_1_1");
+        wb1.setDomain("domain1");
+        workbasketService.updateWorkbasket(wb1);
+        Workbasket wb2 = workbasketService.getWorkbasketByKey("USER_2_1");
+        wb2.setDomain("domain2");
+        workbasketService.updateWorkbasket(wb2);
```
Comment (holgerhagen) [inline lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java]: bitte nicht tun

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java [hunk 2]
```diff
@@ -400,6 +403,102 @@ public void testThrowsExceptionIfMandatoryPrimaryObjectReferenceIsNotSetOrIncomp
...
+
+        assertNotNull(createdTask);
+        assertEquals(workbasket.getDomain(), createdTask.getDomain());
+    }
+
+    @WithAccessId(
+            userName = "user_1_1",
+            groupNames = { "group_1" })
+    @Test
+    public void testDomainChangingWhenTransferTask()
```
[the remark this answers is not in the document]
Comment (BVier) [reply lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java]: Done

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java (hunk 1 above)
Comment (BVier) [reply lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java]: Done

File: lib/taskana-core/src/main/java/pro/taskana/TaskQuery.java [hunk 3]
```diff
@@ -65,6 +65,15 @@
...
     TaskQuery workbasketKeyIn(String... workbasketKeys) throws NotAuthorizedException;
 
+    /**
+     * Add your domain to the query.
+     *
+     * @param domain
+     *            the domain as String
+     * @return the query
+     */
+    TaskQuery domain(String... domain);
```
[the remark this answers is not in the document]
Comment (BVier) [reply lib/taskana-core/src/main/java/pro/taskana/TaskQuery.java]: done

[the remark this answers is not in the document]
Comment (BVier) [reply lib/taskana-core/src/main/java/pro/taskana/TaskQuery.java]: please check, if everything is fine.
I just refactored and build.
But there shouldn't be big problems

**Pull request merged**